### PR TITLE
Release/sdk 0.5.2

### PR DESCRIPTION
**Pull request by PierreJeanjacquot**
### Added

- Added `workerpool` option for `processProtectedData` to override the workerpool to use
- Added a dockerized local stack for testing

### Changed

- Run tests on a local stack forked from bellecour
- Support ENS names for `owner` option of `fetchProtectedData`
- Fixed a bug that allowed `"any"` to be passed as `newOwner` to `transferOwnership`
- Fixed a bug that allowed `"any"` to be passed as `protectedData` to `processProtectedData`
- Fixed a bug that allowed `"any"` to be passed as `app` to `processProtectedData`
- Fixed a bug that may cause the user to pay gas fees when creating a protected data
- Fixed a bug that may cause incorrect requestorder max prices in `processProtectedData`
- Fixed grantAccess to prevent address 0 to be used for authorizedApp
- Changed URL validation to be more permissive

### CODE REVIEW

File: packages/sdk/package.json [hunk 1]
```diff
@@ -45,7 +47,7 @@
     "debug": "^4.3.4",
     "ethers": "^6.8.1",
     "graphql-request": "^6.0.0",
-    "iexec": "^8.5.0",
+    "iexec": "^8.6.0",
```
Comment (cedric25) [inline packages/sdk/package.json:50]: Do we need to add a quick note on the changelog for this iexec-sdk bump?

File: packages/sdk/src/dataProtector/fetchGrantedAccess.ts [hunk 2]
```diff
@@ -15,6 +16,8 @@ export const fetchGrantedAccess = async ({
   protectedData = 'any',
   authorizedApp = 'any',
   authorizedUser = 'any',
+  isAppStrict = false,
+  isUserStrict = false,
```
Comment (cedric25) [inline packages/sdk/src/dataProtector/fetchGrantedAccess.ts:20]: We might need to mention these 2 new input parameters to the changelog?

File: packages/sdk/src/dataProtector/grantAccess.ts [hunk 3]
```diff
@@ -52,6 +54,20 @@ export const grantAccess = async ({
...
+      throw new ValidationError('authorizedApp ENS name is not valid');
+    }
+    vAuthorizedApp = resolved.toLowerCase();
+  }
+
+  if (vAuthorizedApp === ZeroAddress) {
+    throw Error(
+      `Forbidden to use ${ZeroAddress} as authorizedApp, this would give access to any app`
+    );
+  }
```
Comment (cedric25) [inline packages/sdk/src/dataProtector/grantAccess.ts:69]: Should we mention this check in the changelog?

File: packages/sdk/src/dataProtector/fetchGrantedAccess.ts (hunk 2 above)
Comment (PierreJeanjacquot) [reply packages/sdk/src/dataProtector/fetchGrantedAccess.ts:20]: yes good catch

File: packages/sdk/package.json (hunk 1 above)
Comment (PierreJeanjacquot) [reply packages/sdk/package.json:50]: well... we didn't for the other upgrades of the iexec SDK, not sure what to do...

**Review by cedric25 — APPROVED** [review]

File: packages/sdk/package.json (hunk 1 above)
Comment (cedric25) [reply packages/sdk/package.json:50]: No problem then, I wasn't sure :)

**Pull request merged**